src/main/java/org/commcare/formplayer/configuration/WebSecurityConfig.java

```diff
@@ -70,11 +75,18 @@ public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
         // configure auth filters
         http.addFilterAt(getHmacAuthFilter(), UsernamePasswordAuthenticationFilter.class);
         http.addFilterAfter(getSessionAuthFilter(authenticationManager), HmacAuthFilter.class);
-        http.csrf()
-                .csrfTokenRepository(CookieCsrfTokenRepository.withHttpOnlyFalse())
-                .ignoringRequestMatchers(new RequestHeaderRequestMatcher(Constants.HMAC_HEADER));
-        http.cors();
 
+        // By setting the csrfRequestAttributeName to null, the CsrfToken must first be loaded to determine what
+        // attribute name to use. This causes the CsrfToken to be loaded on every request.
+        CsrfTokenRequestAttributeHandler requestHandler = new CsrfTokenRequestAttributeHandler();
+        requestHandler.setCsrfRequestAttributeName(null);
```

Comment:
Noting that this change was in response to a change made in Spring 3 that [defers the csrf tokens lookup](https://docs.spring.io/spring-security/reference/5.8/migration/servlet/exploits.html#_defer_loading_csrftoken) until needed optimising reads of `HttpSession`. Although that breaks all requests to FP as 403 error due to csrf token coming as `undefined` in request headers. In response, I first implemented the opt out steps mentioned [here](https://docs.spring.io/spring-security/reference/5.8/migration/servlet/exploits.html#_i_am_using_a_single_page_application_with_cookiecsrftokenrepository), that didn't solve the problem completely (Half the times requests were with csrf `undefined` and got errored with 403 while on trying again the csrf was there in request payloads, but I could not get to the bottom of why it was happening).  Ultimately I decided to completely opt out of deferred tokens behaviour by follow the steps listed [here](https://docs.spring.io/spring-security/reference/5.8/migration/servlet/exploits.html#_i_need_to_opt_out_of_deferred_tokens_for_another_reason).